src/controller/CHIPDeviceController.h

```diff
@@ -229,17 +219,31 @@ class DLL_EXPORT DeviceController : public Messaging::ExchangeDelegate,
      */
     CHIP_ERROR GetDevice(NodeId deviceId, Device ** device);
 
+    /**
+     *   This function returns true if the device corresponding to `deviceId` has previously been commissioned
+     *   on the fabric.
+     */
+    bool DoesDevicePairingExist(const PeerId & deviceId);
+
+    /**
+     *   This function finds the device corresponding to deviceId, and establishes a secure connection with it.
+     *   Once the connection is successfully establishes (or if it's already connected), it calls `onConnectedDevice`
+     *   callback. If it fails to establish the connection, it calls `onError` callback.
```

Comment:
.. These callbacks must not be already registered with a source (i.e., if they were used previously, they must be canceled).